File: src/frontend/src/app/components/atendimento/agendamento/agenda/model/agenda-dto.ts
import { Exclude, Expose } from 'class-transformer';

@Exclude()
export class AgendaDTO {
  @Expose() id?: string;
  @Expose() nome?: string;
  @Expose() profissionalId?: string;
  @Expose() profissionalNome?: string;
  @Expose() setorId?: string;
  @Expose() setorNome?: string;
  @Expose() ativo?: boolean;
  @Expose() createdAt?: string;
  @Expose() updatedAt?: string;
  @Expose() createdBy?: string;
  @Expose() updatedBy?: string;
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/agendamento/agenda/dto/AgendaDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.agendamento.agenda.dto;

import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class AgendaDTO implements DTO {
    private UUID id;
    private String nome;
    private UUID profissionalId;
    private String profissionalNome;
    private UUID setorId;
    private String setorNome;
    private Boolean ativo;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}
